### ISSUE TITLE
Wrong SQL generated for IIF() emulation in MySQL

### ISSUE THREAD

**Issue by matthijs-oosterhoff** [jOOQ/jOOQ#10939]
### Expected behavior
Using [DSL.iff](https://www.jooq.org/doc/3.14/manual/sql-building/column-expressions/general-functions/iif-function/) should result in a MySQL [IF](https://dev.mysql.com/doc/refman/8.0/en/if.html)

### Actual behavior
The generated query contains a call to the non-existing `iff` function, resulting in the following (MySQL) error:
```
FUNCTION <mydbschema>.iif does not exist
``` 

### Steps to reproduce the problem
- If the problem relates to code generation, please post your code generation configuration
** Gradle configuration snippet **
```
jooq {
    configurations {
        appDb {
            generateSchemaSourceOnCompilation = jooqGenerateSchemaSourceOnCompilation
            generationTool {
                jdbc {
                    driver = 'com.mysql.cj.jdbc.Driver'
                    url = dbConf.url
                    user = dbConf.user
                    password = dbConf.password
                }
                generator {
                    name = 'org.jooq.codegen.DefaultGenerator'
                    database {
                        name = 'org.jooq.meta.mysql.MySQLDatabase'
                        inputSchema = dbConf.name
                        includes = '.*'
                        excludes = 'flyway.*'
                    }
                    generate {
                        pojos = true
                        deprecated = false
                        records = true
                        immutablePojos = true
                    }
                    target {
                        packageName = 'nl.shockmedia.ws.' + dbConf.moduleName + '.generated.db.' + dbConf.moduleName
                    }
                }
            }
        }
    }
}
```

- A complete set of DDL statements can help re-create the setup you're having
- An MCVE can be helpful to provide a complete reproduction case: https://github.com/jOOQ/jOOQ-mcve

### Versions

- jOOQ: 3.14.3
- Java: OpenJDK 11.0.9
- Database (include vendor): MySQL Ver 8.0.22-0ubuntu0.20.04.2 for Linux on x86_64 ((Ubuntu))
- OS: Ubuntu 20.04.1 LTS
- JDBC Driver (include name if inofficial driver): mysql-connector-java 8.0.22

**Comment by lukaseder** [comment]
Thanks for your report. I can confirm, it seems that support for this function has been added without any new integration tests. Will fix immediately.

**Comment by lukaseder** [comment]
The implementation of https://github.com/jOOQ/jOOQ/issues/10160 was incomplete. We should probably treat the two functions (MySQL style `IF()` and T-SQL style `IIF()`) the same way.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.15.0 and 3.14.4 (#10942). For the time being, please use `DSL.if_()` in MySQL

**Comment by matthijs-oosterhoff** [comment]
Thanks for the quick response! 👍
